checks/evaluation/branch_protection.go

```diff
@@ -103,77 +103,37 @@ func BranchProtection(name string, dl checker.DetailLogger,
 	}
 }
 
-func computeNonAdminBasicScore(scores []levelScore) int {
-	score := 0
-	for i := range scores {
-		s := scores[i]
-		score += s.scores.basic
-	}
-	return score
-}
-
-func computeNonAdminReviewScore(scores []levelScore) int {
-	score := 0
-	for i := range scores {
-		s := scores[i]
-		score += s.scores.review
-	}
-	return score
-}
-
-func computeAdminReviewScore(scores []levelScore) int {
-	score := 0
-	for i := range scores {
-		s := scores[i]
-		score += s.scores.adminReview
-	}
-	return score
-}
-
-func computeNonAdminThoroughReviewScore(scores []levelScore) int {
-	score := 0
-	for i := range scores {
-		s := scores[i]
-		score += s.scores.thoroughReview
-	}
-	return score
-}
-
-func computeAdminThoroughReviewScore(scores []levelScore) int {
-	score := 0
-	for i := range scores {
-		s := scores[i]
-		score += s.scores.adminThoroughReview
-	}
-	return score
-}
-
-func computeNonAdminContextScore(scores []levelScore) int {
-	score := 0
-	for i := range scores {
-		s := scores[i]
-		score += s.scores.context
-	}
-	return score
-}
-
-func computeCodeownerThoroughReviewScore(scores []levelScore) int {
-	score := 0
-	for i := range scores {
-		s := scores[i]
-		score += s.scores.codeownerReview
+func sumUpScoreForTier(tier int, scoresData []levelScore, dl checker.DetailLogger) int {
+	sum := 0
+	for i := range scoresData {
+		score := scoresData[i]
+		switch tier {
+		case 1:
+			sum += score.scores.basic
+		case 2:
+			sum += score.scores.review + score.scores.adminReview
+		case 3:
+			sum += score.scores.context
+		case 4:
+			sum += score.scores.thoroughReview + score.scores.codeownerReview
+		case 5:
+			sum += score.scores.adminThoroughReview
+		default:
+			debug(dl, true, "Function sumUpScoreForTier called with the invalid parameter: '%d';"+
+				"BranchProtection score won't be accurate.", tier)
+		}
 	}
-	return score
+	return sum
 }
 
-func noarmalizeScore(score, max, level int) float64 {
+func normalizeScore(score, max, level int) float64 {
 	if max == 0 {
 		return float64(level)
 	}
 	return float64(score*level) / float64(max)
 }
 
-func computeScore(scores []levelScore) (int, error) {
+func computeFinalScore(scores []levelScore, dl checker.DetailLogger) (int, error) {
```

Comment:
mentioned elsewhere. changes likely coming to this code before EOY. so if all of this refactor  is one clean commit, either revert it or understand it will likely change again